syntax = "proto3";

package auth;

service AuthService {
    rpc Register (RegisterRequest) returns (RegisterResponse);
    rpc Login (LoginRequest) returns (LoginResponse);
    rpc ValidateToken (TokenRequest) returns (TokenResponse);
}

message RegisterRequest {
    string username = 1;
    string password = 2;
    string email = 3;
    string role = 4; // Admin, Operador, Supervisor
}

message RegisterResponse {
    bool success = 1;
    string message = 2;
    string user_id = 3;
}

message LoginRequest {
    string username = 1;
    string password = 2;
}

message LoginResponse {
    bool success = 1;
    string message = 2;
    string token = 3;
    string role = 4;
}

message TokenRequest {
    string token = 1;
}

message TokenResponse {
    bool valid = 1;
    string role = 2;
    string user_id = 3;
}